### DISCUSSION TITLE
WebRTC: Method not allowed

### DISCUSSION THREAD

**Discussion by ibrewster** [bluenviron/mediamtx#3564] [Questions]
### Question

When trying to connect to mediamtx via WebRTC, I'm getting a "405 Method Not Allowed error". The relevant portion of the debug log is shown below. This is the current version of mediamtx (1.8.4), and trying to connect from VLC. mediamtx is running directly on my Ubuntu machine, same lan and subnet, no firewall or proxy involved.

I did look at the portion of the readme regarding WebRTC issues, but most of that seems to be if there is NAT, Firewall, proxy, or docker involved, none of which is the case. One thing I did try was adding a local TCP listener, but I can't seem to connect on that port.

This error is when trying to connect with the /whep at the end, without it in a web browser the logs look fine, until I get a `closed: deadline exceeded while waiting connection` error in the logs - and nothing ever loads in the browser.

```
2024/07/19 07:48:05 DEB [WebRTC] [conn 10.x.x.x:53471] [c->s] GET /drivecam/whep HTTP/1.1
Host: myhost.mydomain.com:8889
Accept: */*
Connection: close
Icy-Metadata: 1
Range: bytes=0-
User-Agent: Lavf/60.16.100


2024/07/19 07:48:05 DEB [WebRTC] [conn 10.x.x.x:53471] [s->c] HTTP/1.1 405 Method Not Allowed
Access-Control-Allow-Credentials: true
Access-Control-Allow-Origin: *
Content-Type: application/json; charset=utf-8
Server: mediamtx
```

**Comment by aler9** [comment]
Hello, VLC doesn't support WebRTC or WHEP. Suggested protocols are listed in the README:

https://github.com/bluenviron/mediamtx?tab=readme-ov-file#vlc

For any other info, ask in VLC forums.

Regarding the `closed: deadline exceeded while waiting connection` error, remediations are in the README:

https://github.com/bluenviron/mediamtx?tab=readme-ov-file#solving-webrtc-connectivity-issues
